lib/project_types/extension/loaders/specification_handler.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Extension
+  module Loaders
+    module SpecificationHandler
+      def self.load(project:, context:)
+        identifier = project.specification_identifier
+        Models::LazySpecificationHandler.new(identifier) do
+          specifications = Models::Specifications.new(
+            fetch_specifications: Tasks::FetchSpecifications.new(api_key: project.app.api_key, context: context)
+          )
+
+          unless specifications.valid?(identifier)
+            context.abort(context.message("errors.unknown_type", project.specification_identifier))
```

Comment:
Even though we have several aborts across the codebase, we should move away from that pattern and instead raise an abort error that is handled and formatted. I recommend turning this one into:

```rb
raise ShopifyCLI::Abort, context.message("errors.unknown_type", project.specification_identifier)
```